packages/protocol/test/tokenvault/ERC20Vault.t.sol

```diff
@@ -463,4 +520,162 @@ contract TestERC20Vault is TaikoTest {
+                symbol: "ERC20TT2",
+                name: "ERC20 Test token2"
+            }),
+            address(stETH)
+        );
+
+        vm.warp(block.timestamp + 91 days);
+
+        // usdc is already blacklisted!
+        vm.expectRevert(ERC20Vault.VAULT_BTOKEN_BLACKLISTED.selector);
+        erc20Vault.changeBridgedToken(
+            ERC20Vault.CanonicalERC20({
+                chainId: 1,
+                addr: address(erc20),
+                decimals: 18,
+                symbol: "ERC20TT",
+                name: "ERC20 Test token"
+            }),
+            address(usdc)
+        );
+
+        // We cannot use stETH for erc20 (as it is used in connection with another token)
+        vm.expectRevert(ERC20Vault.VAULT_INVALID_NEW_BTOKEN.selector);
+        erc20Vault.changeBridgedToken(
+            ERC20Vault.CanonicalERC20({
+                chainId: 1,
+                addr: address(erc20),
+                decimals: 18,
+                symbol: "ERC20TT",
+                name: "ERC20 Test token"
+            }),
+            address(stETH)
+        );
+
+        vm.stopPrank();
+    }
+
+    function test_20Vault_to_string() public {
+        vm.startPrank(Alice);
+
+        (, bytes memory symbolData) =
+            address(weirdNamedToken).staticcall(abi.encodeCall(INameSymbol.symbol, ()));
+        (, bytes memory nameData) =
+            address(weirdNamedToken).staticcall(abi.encodeCall(INameSymbol.name, ()));
+
+        string memory decodedSymbol = LibBytes.toString(symbolData);
+        string memory decodedName = LibBytes.toString(nameData);
+
+        assertEq(decodedSymbol, "123456abcdefgh");
+        assertEq(decodedName, "");
+
+        vm.stopPrank();
+    }
+
+    function test_20Vault_deploy_erc20_with_no_name() public {
+        vm.startPrank(Alice);
+
+        vm.chainId(destChainId);
+
+        uint64 amount = 1;
+
+        destChainIdBridge.setERC20Vault(address(destChainIdERC20Vault));
+
+        address bridgedAddressBefore =
+            destChainIdERC20Vault.canonicalToBridged(srcChainId, address(erc20));
+        assertEq(bridgedAddressBefore == address(0), true);
+
+        // todo: dantaik
+        // So this token (weirdNamedToken) has no name (""), as name() and symbol() is optional. And
+        // there is an issue with that.
+        // The following 2 scenarios would create a revert on the "destination" chain:
+        // A: when there is no name (also applies to symbol), we will use that exact same empty
+        // string decoded "" (LibBytes()).
+        // B: if the name() (also applies to symbol) low level call would fail, but we also use
+        // default "" (Base vault line 78).
+        //
+        // In these 2 scenarios, the LibBridgedToken.validateInputs(_srcToken, _srcChainId, _symbol,
+        // _name) -> Would fail.
+        // Either do not invalidate on empty string OR use other than "" as an error default retVal
+        // (in LibBytes) or default value.
```

Comment:
@dantaik found a bug, please review the comment.

Reply:
I prefer remove the name and symbol validation logics.